tests/conftest.py

```diff
@@ -29,41 +29,53 @@ def node(host, request):
         'mimic': 13,
         'dev': 99
     }
-    if not request.node.get_marker(node_type) and not request.node.get_marker('all'):  # noqa E501
-        pytest.skip("Not a valid test for node type: %s" % node_type)
 
-    if request.node.get_marker("no_lvm_scenario") and lvm_scenario:
+    # capture the initial/default state
+    valid_markers = []
+    for marker in request.node.iter_markers():
+        if marker.name not in group_names and marker.name != 'all':
+            valid_markers.append(False)
+        else:
+            valid_markers.append(True)
+        # Check if any of the markers. This is tricky to do with all the negations, let me think a bit here
```

Comment:
This comment looks like it could be removed or reworded. I think we want to say something like:

"Check if any markers on the test method exist in the nodes group_names. If they do not, this test is not valid for the node being tested."